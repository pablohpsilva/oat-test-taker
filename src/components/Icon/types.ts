export default interface IIcon {
    alt: string
    src: string
    size?: number
}
