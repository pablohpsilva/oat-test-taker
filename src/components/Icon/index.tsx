import React, { FC } from 'react'

import IIcon from './types'

const Icon: FC<IIcon> = ({ src, alt = 'Icon...', size = 24 }) =>
    src && (
        <picture>
            <img src={src} alt={alt} height={size} />
        </picture>
    )

export default Icon
