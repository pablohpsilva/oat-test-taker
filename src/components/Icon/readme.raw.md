# Icon

## Props

| Property | Description                 | Type     | Default     |
| -------- | --------------------------- | -------- | ----------- |
| `alt`    | Alt text for accessibility. | `string` | `undefined` |
| `src`    | Icon image source           | `string` | `undefined` |
| `size`   | Icon size                   | `number` | `undefined` |
