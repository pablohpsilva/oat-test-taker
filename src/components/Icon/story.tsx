import React from 'react'
import { storiesOf } from '@storybook/react'
import { text, number } from '@storybook/addon-knobs'

import { Icon } from '@/components'
import readme from './readme.raw.md'

const IconComp = ({
    src = text('src', 'https://placeimg.com/600/300/any'),
    alt = text('alt', 'https://placeimg.com/600/300/any'),
    size = number('size', 30)
}) => <Icon src={src} alt={alt} size={size} />

const opts = {
    notes: { markdown: readme }
}

storiesOf('Components|Icon', module)
    .addDecorator(storyFn => <div>{storyFn()}</div>)
    .add('Example', () => <IconComp />, opts)
