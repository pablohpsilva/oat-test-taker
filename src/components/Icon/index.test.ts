import React from 'react'
import Icon from './'

test('should render the Icon component', async () => {
    const wrapper = shallow(<Icon />)

    expect(wrapper.first().hasClass('Icon')).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()
})
