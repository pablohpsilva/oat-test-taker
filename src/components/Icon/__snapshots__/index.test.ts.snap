// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Icon component 1`] = `false`;

exports[`should render the Icon component 2`] = `""`;
